src/shared/components/FunctionDetailsPanel/infrastructure/useFetchFunctionsDetails.ts

```diff
@@ -35,15 +35,33 @@ export function useFetchFunctionsDetails({
   stacktrace,
   isGitHubLogged,
 }: FetchParams): FetchResponse {
-  // we get the timerange that was used for fetching the main timeline and flamegraph data to ensure data consistency
-  // (see https://github.com/grafana/pyroscope-squad/issues/131)
-  const [start, end] = timelineAndProfileApiClient.getLastTimeRange();
+  let start = 0;
+  let end = 0;
+  let timeRangeError: Error | null = null;
 
-  const { isFetching, error, data } = useQuery({
+  try {
+    // We get the time range that was used for fetching the main timeline and
+    // flame graph data to ensure data consistency.
+    //
+    // see https://github.com/grafana/pyroscope-squad/issues/131
+    [start, end] = timelineAndProfileApiClient.getLastTimeRange();
+  } catch (e) {
+    timeRangeError = e as Error;
+  }
+
+  const {
+    isFetching,
+    error: queryError,
+    data,
+  } = useQuery({
     enabled: Boolean(profileMetricId && labelsSelector && stacktrace.length > 0),
     // eslint-disable-next-line @tanstack/query/exhaustive-deps
     queryKey: ['function-details', profileMetricId, labelsSelector, start, end, stacktrace, isGitHubLogged],
     queryFn: async () => {
+      if (timeRangeError) {
```

Comment:
This should be added to the `enabled` option and not here (see [my comment](https://github.com/grafana/explore-profiles/pull/25/files#r1673652705))

Reply:
My bad! fixed in https://github.com/grafana/explore-profiles/pull/25/commits/c397c75b7c37252245a90b244d3c16ca1a84ca15